controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -341,6 +351,14 @@ func (sp *servicePublisher) newPortPublisher(srcPort Port) *portPublisher {
 	return port
 }
 
+func (sp *servicePublisher) metricsLabels(port Port) prometheus.Labels {
+	return prometheus.Labels{
+		"namespace": sp.id.Namespace,
+		"service":   sp.id.Name,
+		"port":      strconv.Itoa(int(port)),
```

Comment:
assuming this label set is coupled to labels defined in `newEndpointsMetricsVecs()`, how about making it explicit...

```diff
diff --git a/controller/api/destination/watcher/endpoints_watcher.go b/controller/api/destination/watcher/endpoints_watcher.go
index 1fb9d4f6..4bdc095e 100644
--- a/controller/api/destination/watcher/endpoints_watcher.go
+++ b/controller/api/destination/watcher/endpoints_watcher.go
@@ -352,11 +352,7 @@ func (sp *servicePublisher) newPortPublisher(srcPort Port) *portPublisher {
 }
 
 func (sp *servicePublisher) metricsLabels(port Port) prometheus.Labels {
-	return prometheus.Labels{
-		"namespace": sp.id.Namespace,
-		"service":   sp.id.Name,
-		"port":      strconv.Itoa(int(port)),
-	}
+	return endpointsLabels(sp.id.Namespace, sp.id.Name, strconv.Itoa(int(port)))
 }
 
 /////////////////////
diff --git a/controller/api/destination/watcher/prometheus.go b/controller/api/destination/watcher/prometheus.go
index eab99ab3..6fd276d5 100644
--- a/controller/api/destination/watcher/prometheus.go
+++ b/controller/api/destination/watcher/prometheus.go
@@ -57,8 +57,25 @@ func newMetricsVecs(name string, labels []string) metricsVecs {
 	}
 }
 
+func endpointsLabels(namespace, service, port string) prometheus.Labels {
+	return prometheus.Labels{
+		"namespace": namespace,
+		"service":   service,
+		"port":      port,
+	}
+}
+
+func endpointsLabelNames() []string {
+	labelNames := []string{}
+	for name := range endpointsLabels("", "", "") {
+		labelNames = append(labelNames, name)
+	}
+
+	return labelNames
+}
+
 func newEndpointsMetricsVecs() endpointsMetricsVecs {
-	labels := []string{"namespace", "service", "port"}
+	labels := endpointsLabelNames()
 	vecs := newMetricsVecs("endpoints", labels)
 
 	pods := promauto.NewGaugeVec(
```